network/src/peers/peer/connector.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    io::{Error as IoError, ErrorKind},
+    time::Duration,
+};
+
+use futures::{select, FutureExt};
+use snarkvm_dpc::Storage;
+use tokio::{net::TcpStream, sync::mpsc};
+
+use super::PeerAction;
+use crate::{stats, NetworkError, Node, Peer, PeerEvent, PeerEventData, PeerHandle, Version};
+
+use super::network::PeerIOHandle;
+
+const CONNECTION_TIMEOUT_SECS: u64 = 3;
+
+impl Peer {
+    async fn inner_connect(&mut self, our_version: Version) -> Result<PeerIOHandle, NetworkError> {
```

Comment:
Can we add some comments defining the difference between an inner_connect and a connect?

Also please move all private functions to the end of the impl section

Reply:
this is a fairly common pattern to be able to use try operators (`?`), since Rust only lets us use them at a function-return level scope. there isn't a difference other than one handles errors more freely (versus `connect` which has very verbose error handling)